### ISSUE TITLE
Multiple connection support

### ISSUE THREAD

**Issue by lafriks** [go-rel/rel#226]
It would be nice to have support to provide multiple connections with role write or read-only so that it would possible to use for example with PostgreSQL cluster that has single master and multiple read only nodes.

**Comment by Fs02** [comment]
> Rel would then send write operations to master and select operations to one of slaves

In addition to this, it'll be good if REL supports query option to allow reading from master

**Comment by Fs02** [comment]
do you have another idea?

**Comment by lafriks** [comment]
if Query/Aggregate is run in transaction or Query has `rel.ForUpdate` it must also use primary

**Comment by lafriks** [comment]
I mean if I start transaction and do insert/update etc any SELECTS that are done within that transaction should be done within same transaction using same connection

**Comment by Fs02** [comment]
yes, correct, that exactly what will happen since `Begin` method in Adapter is actually returns another adapter with transaction connection, so it should be covered 👍
